From c5e3011b7364729fa2cd4f11761bf1f001a931a4 Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Tue, 22 May 2018 20:45:45 +0200
Subject: [PATCH] Do not restart service after unclean exit code

Currently, the vino-server.service has Restart=on-failure, which means
that it is restarted in abnormal cases, but also in case of non-zero
exit code. It is restarted 5 times e.g. in case when X11 is not detected,
which doesn't make sense. Non-zero exit code is used only for states
which won't change with restart (invalid commandline, wayland and some
sanity checks). Change the value to Restart=on-abnormal in order to
prevent the useless restarts and to not spam journal.

https://bugzilla.gnome.org/show_bug.cgi?id=761120
---
 server/vino-server.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/vino-server.service.in b/server/vino-server.service.in
index a48b813..49e9c1f 100644
--- a/server/vino-server.service.in
+++ b/server/vino-server.service.in
@@ -5,4 +5,4 @@ Description=Vino VNC server
 Type=dbus
 BusName=org.gnome.Vino
 ExecStart=@libexecdir@/vino-server
-Restart=on-failure
+Restart=on-abnormal
-- 
2.17.0
